crates/agglayer-clock/src/block.rs

```diff
@@ -219,34 +262,50 @@ where
                     break;
                 }
                 block_result = Self::recv_block(&mut stream) => {
-                    let block = block_result?;
+                    let block = block_result.map_err(|e| {
+                        error!(error = %e, "Failed to receive block from stream");
```

Comment:
@iljakuklic Continuing discussion from [here](https://github.com/agglayer/agglayer/pull/922#discussion_r2240553249). I think in general it is always useful to show debug printout of the errors in the traces, to display all the information related to this error (e.g. source error chain, or some data missed in the display implementation). In this particular case, this is relevant for `BlockClockError::Transport` source error.